libs/filamentapp/src/NativeWindowHelperLinux.cpp

```diff
@@ -32,7 +32,14 @@ void* getNativeWindow(SDL_Window* sdlWindow) {
     }
     else if (wmi.subsystem == SDL_SYSWM_WAYLAND) {
 #if defined(FILAMENT_SUPPORTS_WAYLAND)
-        return wmi.info.wl.display;
+        static struct {
```

Comment:
Will this ever work for multiple windows/multiple displays? It seems that returning a pointer to a static struct is a recipe for subtle bugs somewhere down the line.

Reply:
As-is this supports multi-window/display scenarios.  The display and surface pointer stays constant during the lifecycle of the Wayland client session.

The easiest scenario to switch startup outputs for an existing Wayland based app is to set the `WAYLAND_DISPLAY` environmental variable to the desired display.  Then of course one can programmatically move/set windows to a desired output.  Wayland based compositors are very flexible in this regard.

So with the Filament demo apps I can launch on any output, or move them between outputs.

Reply:
The other very important fact `static` is addressing a problem.  Thread 1 is the owner of the stack memory which the struct points to.  When the VK surface is created which consumes the pointer, it may be on Thread 73.  If static isn't used you will _most likely_ hit a segfault.

Reply:
Yes without static this would not work, even on a single thread.

Reply:
Okay I really want a comment here explaining everything that was said in this thread.